### PR TITLE
Add Palantir Foundry connector docs page

### PR DESCRIPTION
**Pull request by mshafer25**
Hi Power Query team, this is Mike from the Palantir team owning the Power Query connector for Foundry. It exists in the "List of Power Query Connectors" page already, but this PR adds a proper docs page.

Thanks in advance for your review.

### CODE REVIEW

**Comment by Court72** [comment]
@mshafer25 

Thank you for your contribution. Would you take a moment to sign the [Contributor License Agreement (CLA)](https://github.com/MicrosoftDocs/powerquery-docs/pull/417#issuecomment-1552216352)? After the CLA is signed, someone can review your pull request. Thanks!

#label:"aq-pr-triaged"
@MicrosoftDocs/public-repo-pr-review-team

**Comment by mshafer25** [comment]
@microsoft-github-policy-service agree company="Palantir"

**Comment by Court72** [comment]
@bezhan-msft, @DougKlopfenstein 

Can you review the proposed changes? 

When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge.

File: powerquery-docs/connectors/palantir-foundry-datasets.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: Palantir Foundry Power Query
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/palantir-foundry-datasets.md:2]: nit: Palantir Foundry Power Query Connector

File: powerquery-docs/connectors/palantir-foundry-datasets.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+## Capabilities supported
+
+* Import
+* DirectQuery (Power BI only)
+
+## Connect to Palantir Foundry from Power Query Desktop
+
+To connect to Palantir Foundry from Power Query Desktop, take the following steps:
+
+1. Select the **Palantir Foundry** option in the connector selection.
```
Comment (bezhan-msft) [inline powerquery-docs/connectors/palantir-foundry-datasets.md:38]: nit: "Get Data" experience

**Review by bezhan-msft — CHANGES_REQUESTED** [review]
@DougKlopfenstein to also review for style. This can be approved following Doug's review.

File: powerquery-docs/connectors/palantir-foundry-datasets.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+3. Select **Next** to continue.
+4. In **Navigator**, select the data you require, and then select **Transform data**.
+
+   ![Power Query Online Navigator showing the selected dataset.](./media/palantir-foundry-datasets/online-select-data.png)
+
+### Troubleshooting
+
+If you encounter issues connecting to Foundry, refer to the following resources in Palantir Foundry's documentation for troubleshooting steps:
+
+* [Power BI Connector: FAQs](https://www.palantir.com/docs/foundry/analytics-connectivity/power-bi-faqs/)
```
[the remark this answers is not in the document]
Comment (mshafer25) [reply powerquery-docs/connectors/palantir-foundry-datasets.md:76]: @bezhan-msft for the moment the content would also need to exist in that location on the Palantir docs site, so I would be hesitant to replicate the content here and need to keep them in sync. But we are anticipating a refactor of that documentation section, so at that point we could consider moving it to this Power Query docs site instead. Would that work for you?

**Comment by mshafer25** [comment]
@bezhan-msft thanks for your review, I've updated the content per your suggestions and responded to your question above

File: powerquery-docs/connectors/palantir-foundry-datasets.md (hunk 1 above)
Comment (DougKlopfenstein) [reply powerquery-docs/connectors/palantir-foundry-datasets.md]: The "connector" part of the title should be lowercase - Fixed in review

File: powerquery-docs/connectors/palantir-foundry-datasets.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+## Connect to Palantir Foundry from Power Query Desktop
+
+To connect to Palantir Foundry from Power Query Desktop, take the following steps:
+
+1. Select the **Palantir Foundry** option in the connector selection.
+2. In **Connection Settings**, provide the **Base URL** of your Foundry environment. For example, `https://<subdomain>.palantirfoundry.com/`. Optionally, provide a **Dataset RID** and **Branch**.
+
+   ![Enter Palantir Foundry connection settings in Power Query Desktop.](./media/palantir-foundry-datasets/connection-settings.png)
+
+3. Select either the **Import** or **DirectQuery** data connectivity mode.
```
[the remark this answers is not in the document]
Comment (DougKlopfenstein) [reply powerquery-docs/connectors/palantir-foundry-datasets.md:43]: The "DirectQuery" feature is actually one word - fixed in review.

File: powerquery-docs/toc.yml [hunk 5]
```diff
@@ -450,6 +450,8 @@
         href: connectors/opensearch-project.md
       - name: "Oracle database"
         href: connectors/oracle-database.md
+      - name: "Palantir Foundry"
```
[the remark this answers is not in the document]
Comment (DougKlopfenstein) [reply powerquery-docs/toc.yml:453]: This appears to be fixed.

**Review by DougKlopfenstein — APPROVED** [review]

**Comment by DougKlopfenstein** [comment]
#sign-off

**Pull request merged**